modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceTest.java

```diff
@@ -271,39 +269,30 @@ public void testGetGroupRolesAndTeamRoles() throws Exception {
 			companyId, null, excludedRoleNames, roleTypes, 0, groupId,
 			QueryUtil.ALL_POS, QueryUtil.ALL_POS);
 
-		List<Role> expectedRoles = _roleLocalService.getRoles(companyId);
+		List<Role> expectedRoles = ListUtil.filter(
+			_roleLocalService.getRoles(companyId),
+			role -> {
+				if (excludedRoleNames.contains(role.getName()) ||
+					(role.getType() == RoleConstants.TYPE_ACCOUNT) ||
+					(role.getType() == RoleConstants.TYPE_DEPOT) ||
+					(role.getType() == RoleConstants.TYPE_SITE) ||
+					!role.isTeam()) {
```

Comment:
@jeffwu0724 What's the reason to merge `role.isTeam()` check into this condition?

Reply:
Hi @dantewang ,
My first attempt to buy the sapce is return false right after the frist if. But in this way, we do not really reduce the number of lines. 
Therefore, based on my previous change
```
if (!excludedRoleNames.contains(role.getName()) &&
(role.getType() != RoleConstants.TYPE_ACCOUNT) &&
					(role.getType() != RoleConstants.TYPE_DEPOT) &&
					(role.getType() != RoleConstants.TYPE_SITE)) {

					if (role.getType() != RoleConstants.TYPE_PROVIDER) {
						return true;
					}

					if (!role.isTeam()) {
						return false;
					}

					Team team = _teamLocalService.fetchTeam(role.getClassPK());

					if (team == null) {
						return false;
					}

					return team.getGroupId() == groupId;
				}

				return false;
			});
``` 
For both the outer if and if (!role.isTeam()) will return false if we did not match the condition, then I think we can combine them together.

Please correct me if I made any mistakes. Really appreciate your help!

Reply:
@jeffwu0724 I'm just wondering if the order of the conditions matter. The original logic checks the `role.getType()` first, and then `role.isTeam()` related logic. So, if the type is not `TYPE_PROVIDER`, even if the `role` is not `team`, this should return true. I'd say it's better for us to keep the original order.

Also, "buy space" usually means to reduce the levels of indents, so the lines of code don't need to be shorter after "buying" space.